File: competence-database/src/main/scala/uzuzjmd/competence/persistence/dao/Comment.scala
package uzuzjmd.competence.persistence.dao

import uzuzjmd.competence.persistence.abstractlayer.{CompOntologyManager, CompOntologyAccess}
import uzuzjmd.competence.persistence.ontology.CompOntClass
import uzuzjmd.competence.persistence.ontology.CompObjectProperties
import uzuzjmd.competence.persistence.owl.CompOntologyManagerJenaImpl

case class Comment(comp: CompOntologyManager, val text: String, val creator: User = null, val created: java.lang.Long = null, val textReadable: String = null) extends CompetenceOntologyDao(comp, CompOntClass.Comment, text) {

  def DATECRATED = "datecreated"
  def TEXT = "text"

  @Override
  protected def deleteMore() {
    //TODO

  }

  @Override
  protected def persistMore() {
    creator.persist
    createEdgeWith(creator, CompObjectProperties.UserOfComment)
    addDataField(TEXT, identifier)
    if (created == null) {
      addDataField(DATECRATED, System.currentTimeMillis(): java.lang.Long);
    } else {
      addDataField(DATECRATED, created);
    }
  }

  def getFullDao(): Comment = {
    val creatorNew = getAssociatedStandardDaosAsDomain(CompObjectProperties.UserOfComment, classOf[User]).head
    val textNew = getDataField(TEXT)
    val dateCreatedNew = getDataFieldLong(DATECRATED)
    return new Comment(comp, textNew, creatorNew, dateCreatedNew, textNew)
  }
}

File: competence-database/src/main/scala/uzuzjmd/competence/persistence/dao/Comment.java
package uzuzjmd.competence.persistence.dao;

import uzuzjmd.competence.exceptions.NoUserGivenException;
import uzuzjmd.competence.persistence.ontology.Edge;
import uzuzjmd.competence.service.rest.dto.CommentData;

/**
 * Created by dehne on 11.01.2016.
 */
public class Comment extends AbstractComment implements Cascadable {

    public Long dateCreated;
    public String text;

    public Comment(String id) {
        super(id);
    }

    public Comment(String text, User creator, Long dateCreated) {
        super(dateCreated + text);
        this.creator = creator;
    }

    public Long getDateCreated() {
        return dateCreated;
    }

    public void setDateCreated(Long dateCreated) {
        this.dateCreated = dateCreated;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public CommentData getData(){
        CommentData commentData = new CommentData(null, this.getText(), null, null, null, null);
        return commentData;
    }

    @Override
    public void persistMore() throws Exception {
        this.persist();
        if (creator == null) {
            throw new NoUserGivenException();
        }
        createEdgeWith(creator, Edge.UserOfComment);
    }




}
